test/test_response_header.rb

```diff
@@ -49,15 +49,15 @@ def test_integer_key
     server_run app: ->(env) { [200, { 1 => 'Boo'}, []] }
     data = send_http_and_read "GET / HTTP/1.0\r\n\r\n"
 
-    assert_match(/HTTP\/1.1 500 Internal Server Error/, data)
+    assert_match(/HTTP\/1.0 500 Internal Server Error/, data)
```

Comment:
🤔 Huh?

Reply:
Previously it failed on some platforms (non MRI or windows I think). This is the full `data`
```
HTTP/1.0 500 Internal Server Error\r\nContent-Length: 659\r\n\r\nPuma caught this error: undefined method `downcase' for 1:Integer (NoMethodError)\n/Users/vutran/learning/puma/lib/puma/request.rb:610:in `block in str_headers'\n/Users/vutran/learning/puma/lib/puma/request.rb:607:in `each'\n/Users/vutran/learning/puma/lib/puma/request.rb:607:in `str_headers'\n/Users/vutran/learning/puma/lib/puma/request.rb:170:in `prepare_response'\n/Users/vutran/learning/puma/lib/puma/request.rb:131:in `handle_request'\n/Users/vutran/learning/puma/lib/puma/server.rb:431:in `process_client'\n/Users/vutran/learning/puma/lib/puma/server.rb:233:in `block in run'\n/Users/vutran/learning/puma/lib/puma/thread_pool.rb:147:in `block in spawn_thread'HTTP/1.1 500 Internal Server Error\r\n\r\n
```

I guess matching against `Puma caught this error: undefined method `downcase' for 1:Integer` would be better?

Reply:
@nateberkopec I commented on the same thing with the previous attempt: https://github.com/puma/puma/pull/2731/files#diff-732bb402028f0495d2e1bbcdf2abd9deef82f19e1b5f45711e5e32d450656563

We should understand why this is changing, not just change it

Reply:
> This is the full `data`

Why does it say `HTTP/1.1 500 Internal Server Error\r\n\r\n` at the end?

What does it say in master?

Reply:
> Why does it say HTTP/1.1 500 Internal Server Error\r\n\r\n at the end?

That comes from the `client.write_error(500)`. We can either
1. keep that line and change the regex to make the test pass
2. remove that line completely

I went with (1) because I wasn't sure if (2) has any other implications